addons/mrp/models/mrp_production.py

```diff
@@ -1848,7 +1849,7 @@ def _pre_button_mark_done(self):
             return self._action_generate_consumption_wizard(consumption_issues)
 
         quantity_issues = self._get_quantity_produced_issues()
-        if quantity_issues:
+        if quantity_issues and self.picking_type_id.create_backorder == 'ask':
```

Comment:
This is also incorrect. The `_action_generate_backorder_wizard` should check which MOs should be asked to be backordered in the wizard because you can have 2 MO operation types with differing `create_backorder` values. You also need to add in extra logic to correctly handle the `always` case.